back-end/e-learning/app/Http/Controllers/UserController.php

```diff
@@ -31,12 +31,28 @@ public function me()
   {
     return response(auth()->user(), 201);
   }
-  
+
   public function logout()
   {
     auth()->user()->tokens()->delete();
-    return [
+    return response([
       'message' => 'Logout'
-    ];
+    ], 201);
+  }
+
+  public function signup(Request $request)
+  {
+    $user = new User();
+    $user->first_name = $request->first_name;
+    $user->last_name = $request->last_name;
+    $user->email = $request->email;
+    $user->password = Hash::make($request->password);
+    $user->role = 'member';
```

Comment:
You can use `fill` here to reduce number of times of referencing to `$user`

`fill` is like `update` only that it does not update immediately. This is useful if you still want to have a logic to handle when a user is successfully saved or not. Like what you have on `line 51`

example:
```
// fill() will save the values into memory temporarily until the save() function is called for the user
$user->fill([
  'name' => 'Aubrey Vicente',
  'age' => 25
  'email' => 'aubrey@test.com'
]);

// save() will complete the update process of the user above
$user->save();
```

Reply:
I will look into this ma'am